libdnf/advisory/advisory_package.cpp

```diff
@@ -0,0 +1,133 @@
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/advisory/advisory_package.hpp"
+
+#include "advisory_package_private.hpp"
+
+#include "libdnf/logger/logger.hpp"
+#include "libdnf/rpm/nevra.hpp"
+#include "libdnf/utils/utils.hpp"
+
+#include <solv/chksum.h>
+#include <solv/repo.h>
+#include <solv/util.h>
+
+
+namespace libdnf::advisory {
+
+// AdvisoryPackage
+AdvisoryPackage::AdvisoryPackage(AdvisoryPackagePrivate * private_pkg) : private_pkg(private_pkg) {}
+
+AdvisoryPackage::~AdvisoryPackage() = default;
+
+AdvisoryPackage::AdvisoryPackage(const AdvisoryPackage & src) : private_pkg(new AdvisoryPackagePrivate) {
+    *private_pkg = *src.private_pkg;
+}
+AdvisoryPackage::AdvisoryPackage(AdvisoryPackage && src) : private_pkg(new AdvisoryPackagePrivate) {
+    private_pkg.swap(src.private_pkg);
+}
+
+AdvisoryPackage & AdvisoryPackage::operator=(const AdvisoryPackage & src) {
+    *private_pkg = *src.private_pkg;
+    return *this;
+}
+
+AdvisoryPackage & AdvisoryPackage::operator=(AdvisoryPackage && src) noexcept {
+    private_pkg.swap(src.private_pkg);
+    return *this;
+}
+
+
+std::string AdvisoryPackage::get_name() const {
+    return private_pkg->get_name();
+}
+
+std::string AdvisoryPackage::get_version() const {
+    return private_pkg->get_version();
+}
+
+std::string AdvisoryPackage::get_evr() const {
+    return private_pkg->get_evr();
+}
+std::string AdvisoryPackage::get_arch() const {
+    return private_pkg->get_arch();
+}
+AdvisoryId AdvisoryPackage::get_advisory_id() const {
+    return private_pkg->get_advisory_id();
+}
+bool AdvisoryPackage::compareNEVRA(const AdvisoryPackage & first, const AdvisoryPackage & second) {
+    return AdvisoryPackagePrivate::compareNEVRA(*(first.private_pkg.get()), *(second.private_pkg.get()));
+}
+bool AdvisoryPackage::comparePackageNA(const AdvisoryPackage & adv_pkg, const rpm::Package & pkg) {
+    return AdvisoryPackagePrivate::comparePackageNA(*(adv_pkg.private_pkg.get()), pkg);
+}
+
+// AdvisoryPackagePrivate
+AdvisoryPackage::AdvisoryPackagePrivate::AdvisoryPackagePrivate(
+    libdnf::Base & base, AdvisoryId advisory, Id name, Id evr, Id arch, const char * filename)
+    : base(&base)
+    , advisory(advisory)
+    , name(name)
+    , evr(evr)
+    , arch(arch)
+    , filename(filename) {}
+
+std::string AdvisoryPackage::AdvisoryPackagePrivate::get_name() const {
+    auto & solv_sack = base->get_rpm_solv_sack();
+    Pool * pool = solv_sack.p_impl->get_pool();
+    return pool_id2str(pool, name);
+}
+
+std::string AdvisoryPackage::AdvisoryPackagePrivate::get_version() const {
+    auto & solv_sack = base->get_rpm_solv_sack();
+    Pool * pool = solv_sack.p_impl->get_pool();
+    const char * evr_str = pool_id2str(pool, evr);
+    libdnf::rpm::Nevra nevra;
+    nevra.parse(evr_str, libdnf::rpm::Nevra::Form::EVR);
```

Comment:
What about to use solv::get_version(pool, id)? Using a regex is much slower.

Reply:
yes, that is a good idea and I can drop the first commit.